### PR TITLE
Order add update date with time and make order status dynamic/automatic loading

### PR DESCRIPTION
**Pull request by cis-shubham-t**

### CODE REVIEW

File: src/app/core/market/api/bid/bid.service.ts [hunk 1]
```diff
@@ -50,7 +50,8 @@ export class BidService {
 
   search(address: string, type: any): Observable<any> {
     const params = ['search', '*', '*', 'ASC'];
-    return this.market.call('bid', params).map(o => new Bid(o, address, type))
+    return this.market.call('bid', params).distinctUntilChanged()
```
Comment (kewde) [inline src/app/core/market/api/bid/bid.service.ts]: No need for distinctUntilChanged here..

File: src/app/core/util/utils.ts [hunk 2]
```diff
@@ -349,12 +349,13 @@ export const Messages = {
...
   }
 }
+
 export const setOrderKeys = (ord: any, type: string) => {
   // once the order has been accepted then we get status in orderItem
   const status = ord.OrderItem.status ? ord.OrderItem.status : ord.action === 'MPA_REJECT' ? 'REJECTED' : 'BIDDING';
   ord.type = type;
-  ord.added = new DateFormatter(new Date(ord.createdAt)).dateFormatter(true);
-  ord.updated = new DateFormatter(new Date(ord.updatedAt)).dateFormatter(true);
+  ord.added = new DateFormatter(new Date(ord.createdAt)).dateFormatter(false);
```
Comment (kewde) [inline src/app/core/util/utils.ts]: I think you can just omit false here.

File: src/app/market/shared/orders/orders.component.ts [hunk 3]
```diff
@@ -53,16 +57,45 @@ export class OrdersComponent implements OnInit {
...
   }
 
-  loadOrders(): void {
-    this.bid.search(this.profile.address, this.type).take(1).subscribe(orders => {
-      console.log('called >>>>>>>>>>>>>>>>>', orders);
-      orders.orders.reverse();
-      this.orders = orders;
+  callLoadOffers() {
+    this.loadOrders()
+    this.timer = Observable.interval(1000 * 10);
```
Comment (kewde) [inline src/app/market/shared/orders/orders.component.ts]: This is every 10s, not every 30s

File: src/app/market/shared/orders/orders.component.ts [hunk 4]
```diff
@@ -53,16 +57,45 @@ export class OrdersComponent implements OnInit {
...
+        this.callLoadOffers();
       });
   }
 
-  loadOrders(): void {
-    this.bid.search(this.profile.address, this.type).take(1).subscribe(orders => {
-      console.log('called >>>>>>>>>>>>>>>>>', orders);
-      orders.orders.reverse();
-      this.orders = orders;
+  callLoadOffers() {
```
Comment (kewde) [inline src/app/market/shared/orders/orders.component.ts]: initLoopLoadOrders()

File: src/app/market/shared/orders/orders.component.ts [hunk 5]
```diff
@@ -53,16 +57,45 @@ export class OrdersComponent implements OnInit {
...
+          this.orders['orders'].length !== orders.orders.length ||
+          this.hasUpdatedOrders(this.orders['orders'], orders.orders)
+        ) {
+          this.orders = orders;
+        }
+     });
+  }
+
+  hasUpdatedOrders(newOrders: any, oldOrders: any): boolean {
+    return !! (
```
Comment (kewde) [inline src/app/market/shared/orders/orders.component.ts]: !! can be removed?

File: src/app/market/shared/orders/orders.component.ts [hunk 6]
```diff
@@ -53,16 +57,45 @@ export class OrdersComponent implements OnInit {
...
 
+  loadOrders(): void {
+    this.bid.search(this.profile.address, this.type)
+      .take(1)
+      .subscribe(orders => {
+        console.log('called >>>>>>>>>>>>>>>>>', orders);
+        orders.orders.reverse();
+        if (
+          !this.orders ||
+          this.orders['orders'].length !== orders.orders.length ||
```
Comment (kewde) [inline src/app/market/shared/orders/orders.component.ts]:
```
          !this.orders ||
          this.orders['orders'].length !== orders.orders.length ||
```
move both these lines into this.hasUpdatedOrders?

File: src/app/market/shared/orders/orders.component.ts (hunk 3 above)
Comment (cis-shubham-t) [reply src/app/market/shared/orders/orders.component.ts]: @kewde  yes, 30 sec seems like a little late, so I changed it to 10 sec. :)

**Pull request merged**